src/makeInfinityAdder.js

```diff
@@ -5,8 +5,24 @@
  * @return {function}
  */
 
-function makeInfinityAdder() {
-  // write code here
+function makeInfinityAdder(x) {
+  let sum = 0;
+
+  function adder(number) {
+    if (number !== undefined) {
+      sum += number;
+
+      return adder;
+    } else {
```

Comment:
```suggestion
      return adder;
    }
```
Else is redundant here. Your code will not go further after `return` anyway